### DISCUSSION TITLE
Partial layer freezing for fine-tuning to mitigate catastrophic forgetting on dopant elements

### DISCUSSION THREAD

**Discussion by sek1ro-yuzzz** [MDIL-SNU/SevenNet#269] [Q&A]
## Background

I am working on solid-state electrolyte research. I have successfully fine-tuned SevenNet-mf-ompa (modal=mpa) on amorphous solid-state electrolyte systems such as NaTaCl6 and NaNbOCl4, and the model performs well in predicting properties such as energy, forces, and stress.

---

## Problem

Recently, I started studying NaTaCl6 structures doped with various elements (such as Al, Ga, Ti, Zr, Hf, etc.). However, I found that after fine-tuning on the pure NaTaCl₆ system, the model experienced forgetting—it lost its original accuracy for the dopant elements that the pretrained model could describe well (as judged by polyanionic polyhedral volumes, polyhedral bond angle distributions, etc.).

---

## Help I'm Seeking

I am not very familiar with machine learning training techniques, but from what I understand, **partial layer freezing** during fine-tuning might help preserve the pretrained chemical knowledge while only making lightweight corrections to the host system.

I reviewed the codebase and found that currently only `train_shift_scale` and `train_denominator` can be toggled. There doesn't seem to be a built-in option to freeze specific layers (such as the embedding layer or early interaction blocks).

---

## Questions

1. Is there an official or recommended method for partial layer freezing in SevenNet fine-tuning?
2. If not, is it feasible to manually set certain layers to `requires_grad=False` before training? For example:
    
    ```python
    # Freeze embedding and early layers
    for name, param in model.named_parameters():
        if 'onehot_to_feature_x' in name or '0_convolution' in name:
            param.requires_grad = False
    ```
    
3. Are there other recommended strategies to mitigate catastrophic forgetting in this scenario? For instance, using a much lower learning rate?

---

Thank you for developing this excellent tool! Any guidance would be greatly appreciated.

**Comment by YutackPark** [comment]
I'm not familiar with layer freezing but you can implement by yourself and sevennet does not prevent such modifications.

However, I highly recommend you read this paper: https://www.nature.com/articles/s41524-025-01895-w

The author is planning to implement the method to here (the official, latest sevennet repository). But detailed timeline or plan is up to him (@kskjs1203 , please leave us some comment)

**Comment by kskjs1203** [comment]
Thanks for your interest.

Regarding mitigating catastrophic forgetting, we recommend referring to the following work (https://www.nature.com/articles/s41524-025-01895-w).

It is worth noting that the current results in the paper are based on SevenNet-0, and direct application to SevenNet-Omni / SevenNet-mf-ompa is currently under active testing. We are in the process of validating and adapting the method for the Omni/MF variants. Once these tests are completed, we plan to update the official SevenNet repository with the corresponding code and examples.

Until then, while manual layer freezing (e.g., setting requires_grad = False) is technically feasible and not blocked by SevenNet, it may in some cases help reduce forgetting, especially when the fine-tuning target is very close to the pretrained domain and only small, localized corrections are needed. However, in our experience, layer freezing is hard to be a generally reliable strategy for forgetting prevention in universal MLIPs.
